tensorflow/python/keras/layers/convolutional.py

```diff
@@ -231,6 +231,33 @@ def build(self, input_shape):
     if tf_op_name == 'Conv1D':
       tf_op_name = 'conv1d'  # Backwards compat.
 
+    # Check if output shapes are valid
+    # They must not have 0 entries along any dimension
+    if self._channels_first:
```

Comment:
Could you explain why we need to skip checks on the channel axis? Also please consider putting a comment before the if check.

Reply:
That missed that. I'll fix the code and push again. Thanks for the inputs.

Reply:
Also the rest of the arguments for `conv_utils.conv_output_length` don't consider channels and batch size.